### PR TITLE
Avoid deployment-summary related crashes

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

When full deployment information isn't downloaded yet, there can be some `undefined` values where previously there were defined values. This needs to be accounted for everywhere, otherwise crashes occur.

### CODE REVIEW

**Review by philip-cline — APPROVED** [review]
would love to see `nysdot-hopr` incorporated in here but will defer to later reviewers.

**Comment by miles-grant-ibigroup** [comment]
If `nysdot-hopr` is causing conflicts I'd suggest having that PR block this one.

**Review by daniel-heppner-ibigroup — APPROVED** [review]
I don't know what nysdot-hopr is so this is approved.

**Pull request merged**
